### PR TITLE
TDV-310 Improve test execution

### PR DESCRIPTION
**Pull request by ardier16**

### CODE REVIEW

File: test/console-setup.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  )
+}
+
+console.warn = message => {
+  consoleErrors.push(
+    new Error(`console.warn is forbidden during test execution. Message: ${message}`)
+  )
+}
+
+afterEach(() => {
```
Comment (max-nes) [inline test/console-setup.js]: Add a comment to explain what happened

File: karma.conf.js [hunk 2]
```diff
@@ -17,10 +17,12 @@ module.exports = function (config) {
       },
     },
     files: [
+      'test/console-setup.js',
```
Comment (max-nes) [inline karma.conf.js]: disallow-console-calls.js

File: test/console-setup.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+let consoleErrors = []
+
+console.error = e => {
```
Comment (max-nes) [inline test/console-setup.js]: decorate, do not rewrite

File: test/console-setup.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
+let consoleErrors = []
```
Comment (max-nes) [inline test/console-setup.js]: let consoleCallCounter = 0

const errorFn = console.error
console.error = (...args) => {
  consoleCallCounter++
  errorFn(...args)
}

File: CHANGELOG.md [hunk 5]
```diff
@@ -36,6 +36,7 @@ for further information about branching and tagging conventions.
 - Correct base/quote price calculating for Opportunities
 - Custom validation rules for sale's "Soft cap" and "Hard cap" values
 - Custom Movements TopBar for REIT scheme
+- Failing a test if console method was called during test execution
```
Comment (max-nes) [inline CHANGELOG.md]: Update the log after the changes accordingly pls

File: test/console-setup.js (hunk 1 above)
Comment (ardier16) [reply test/console-setup.js]: Added

File: karma.conf.js (hunk 2 above)
Comment (ardier16) [reply karma.conf.js]: Renamed

File: test/console-setup.js (hunk 3 above)
Comment (ardier16) [reply test/console-setup.js]: Decorated

File: test/console-setup.js (hunk 4 above)
Comment (ardier16) [reply test/console-setup.js]: Integrated

File: CHANGELOG.md (hunk 5 above)
Comment (ardier16) [reply CHANGELOG.md]: Updated

**Pull request merged**